occurrences/models.py

```diff
@@ -506,3 +511,25 @@ def decline(self, custom_message=None):
             custom_message=custom_message,
             enrolment=self,
         )
+
+    def create_cancellation_token(self):
+        """
+        Create a cancellation verification token for an enrolment.
+        """
+        return VerificationToken.objects.create_token(
+            self, self.person.user, VerificationToken.VERIFICATION_TYPE_CANCELLATION
+        )
+
+    def create_cancellation_url(self):
+        """
+        Create an URL that can be used to cancel enrolment.
+        URL needs an cancellation token, so one will be created
+        and any active cancellation tokens related to enrolment
+        will be set in_active.
+        """
+        return VerificationToken.objects.deactivate_and_create_token(
+            self,
+            self.person.user,
+            verification_type=VerificationToken.VERIFICATION_TYPE_CANCELLATION,
+            deactivate_all_types_of_token=False,
+        )
```

Comment:
Cancellation urls can be created via enrolment instance. It will always create a new cancellation token and deactivate all the old active cancellation tokens.

Reply:
now it returns the token, but later it will return the actual link!